lib/improver/spotdata/neighbour_finding.py

```diff
@@ -0,0 +1,445 @@
+        ''' Return local grid neighbour with minimum vertical displacement'''
+        return self.minimum_height_error_neighbour(cube, sites,
+                                                   relative_z=None,
+                                                   **kwargs)
+
+    def min_dz_biased_above(self, cube, sites, **kwargs):
+        '''
+        Return local grid neighbour with minimum vertical displacement,
+        biased to select grid points above the site altitude.
+
+        '''
+        return self.minimum_height_error_neighbour(cube, sites,
+                                                   relative_z='above',
+                                                   **kwargs)
+
+    def min_dz_biased_below(self, cube, sites, **kwargs):
+        '''
+        Return local grid neighbour with minimum vertical displacement,
+        biased to select grid points below the site altitude.
+
+        '''
+        return self.minimum_height_error_neighbour(cube, sites,
+                                                   relative_z='below',
+                                                   **kwargs)
+
+    def min_dz_land_no_bias(self, cube, sites, **kwargs):
+        '''
+        Return local grid neighbour with minimum vertical displacement.
+        Require land point neighbour if site is a land point.
+
+        '''
+        return self.minimum_height_error_neighbour(cube, sites,
+                                                   relative_z=None,
+                                                   land_constraint=True,
+                                                   **kwargs)
+
+    def min_dz_land_biased_above(self, cube, sites, **kwargs):
+        '''
+        Return local grid neighbour with minimum vertical displacement,
+        biased to select grid points above the site altitude.
+        Require land point neighbour if site is a land point.
+
+        '''
+        return self.minimum_height_error_neighbour(cube, sites,
+                                                   relative_z='above',
+                                                   land_constraint=True,
+                                                   **kwargs)
+
+    def min_dz_land_biased_below(self, cube, sites, **kwargs):
+        '''
+        Return local grid neighbour with minimum vertical displacement,
+        biased to select grid points below the site altitude.
+        Require land point neighbour if site is a land point.
+
+        '''
+        return self.minimum_height_error_neighbour(cube, sites,
+                                                   relative_z='below',
+                                                   land_constraint=True,
+                                                   **kwargs)
+
+
+def apply_bias(relative_z, dzs, dz_nearest, altitude, orography, i, j):
+    '''
+    Bias neighbour selection to look for grid points with an
+    altitude that is above or below the site if relative_z is
+    not None.
+
```

Comment:
This function was a horrible mess and giving the wrong results :-(
Fixed now and beautifully commented :-)